akka-http-core/src/main/scala/akka/http/javadsl/settings/SettingsCompanion.scala

```diff
@@ -10,7 +10,7 @@ trait SettingsCompanion[T] {
    *
    * Java API
    */
-  final def create(system: ActorSystem): T = create(system.settings.config)
+  def create(system: ActorSystem): T
```

Comment:
Hm, technically this is a breaking change (to make it abstract), not really aimed for anyone to extend it other than us however we missed the chance to mark it as such.

We could have marked this trait `/** INTERNAL API */ private[akka]`, but retroactively doing so is not a good idea I think. WDYT